### ISSUE TITLE
[Getting Started Instructions] Compilation breaks when building OptiML

### ISSUE THREAD

**Issue by radoye** [stanford-ppl/hyperdsl#50]
Hey guys;
I'm trying to build the tools to follow along when reading some of your papers, and maybe play with Delite/Forge. I don't particularly care about OptiML, but I was going through the motions to confirm that everything works.

Some info:
 * Cloned the repository earlier today.
 * The first build passes, though not cleanly (quite a few warnings).
 * The second, ```forge/bin/update ppl.dsl.forge.dsls.optiml.OptiMLDSLRunner OptiML``` does not.
 * I'm following the [getting started](http://stanford-ppl.github.io/Delite/optiml/getting_started.html) instructions.
 * My java is
```
⌂135% [rrs:~/workspace/private/study/classes/delite/hyperdsl] master 2 ± java -version
java version "1.7.0_60"
Java(TM) SE Runtime Environment (build 1.7.0_60-b19)
Java HotSpot(TM) 64-Bit Server VM (build 24.60-b09, mixed mode)
```
 * Any hints? Am I missing a step, did not pay attention to deps/versions, building from a wrong branch?
 * The first (quite a) few messages before it flips out are below.

```
⌂386% [rrs:~/workspace/private/study/classes/delite/hyperdsl] master ± forge/bin/update ppl.dsl.forge.dsls.optiml.OptiMLDSLRunner OptiML
sbt compile (1/4)
[info] Loading project definition from /Users/rrs/workspace/private/study/classes/delite/hyperdsl/project/project
[info] Loading project definition from /Users/rrs/workspace/private/study/classes/delite/hyperdsl/project
[info] Set current project to hyperdsl (in build file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/)
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/}runtime...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/}framework...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/}delite-test...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[success] Total time: 34 s, completed Apr 15, 2015 4:47:53 PM
forge ppl.dsl.forge.dsls.optiml.OptiMLDSLRunner  (2/4)
== Generating DSL with options: {'gen_delite': True, 'verbose': False, 'gen_ident': False, 'build_dir': '/Users/rrs/workspace/private/study/classes/delite/hyperdsl', 'no_blas': False, 'fast_compile': False, 'gen_lib': True}
[forge]: DSL Being Staged:[ppl.dsl.forge.dsls.optiml.OptiMLDSLRunner$] (OptiML)
[forge]: DSL generation complete. Please run publish and compile the generated files against Delite to check for errors.
publish OptiML     (3/4)
== Publishing OptiML to /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published//OptiML/
[forge]: Publishing complete.
~/workspace/private/study/classes/delite/hyperdsl ~/workspace/private/study/classes/delite/hyperdsl
cd published/OptiML/
sbt compile (4/4)
[info] Loading project definition from /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/project
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/project/}optiml-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Compiling 1 Scala source to /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/project/target/scala-2.10/sbt-0.13/classes...
[warn] there were 2 deprecation warning(s); re-run with -deprecation for details
[warn] one warning found
[info] Set current project to OptiML (in build file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/)
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/}OptiML-shared...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading http://repo1.maven.org/maven2/com/google/guava/guava/17.0/guava-17.0.jar ...
[info] 	[SUCCESSFUL ] com.google.guava#guava;17.0!guava.jar(bundle) (1065ms)
[info] Done updating.
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/}OptiML-lib...
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/}OptiML-ident...
[info] Resolving org.scala-lang.virtualized#scala-compiler;2.10.2 ...
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/}OptiML-comp...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading http://repo1.maven.org/maven2/com/googlecode/netlib-java/netlib-java/0.9.3/netlib-java-0.9.3.jar ...
[info] 	[SUCCESSFUL ] com.googlecode.netlib-java#netlib-java;0.9.3!netlib-java.jar (329ms)
[info] downloading http://repo1.maven.org/maven2/net/sourceforge/f2j/arpack_combined_all/0.1/arpack_combined_all-0.1.jar ...
[info] 	[SUCCESSFUL ] net.sourceforge.f2j#arpack_combined_all;0.1!arpack_combined_all.jar (526ms)
[info] Done updating.
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/}OptiML-apps...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Updating {file:/Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/}OptiML...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Compiling 73 Scala sources to /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/shared/target/scala-2.10/classes...
[warn] there were 182 feature warning(s); re-run with -feature for details
[warn] one warning found
[info] Compiling 111 Scala sources to /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/library/target/scala-2.10/classes...
[info] Compiling 109 Scala sources to /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/compiler/target/scala-2.10/classes...
[warn] /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/library/src/optiml/library/classes/Cast.scala:23: abstract type B is unchecked since it is eliminated by erasure
[warn]     __arg0.isInstanceOf[B]
[warn]                        ^
[error] error while loading ForwardTransformer, /Users/rrs/workspace/private/study/classes/delite/hyperdsl/virtualization-lms-core/target/scala-2.10/classes/scala/virtualization/lms/common/ForwardTransformer.class (No such file or directory)
[error] 
[error]      while compiling: /Users/rrs/workspace/private/study/classes/delite/hyperdsl/published/OptiML/library/src/optiml/library/classes/Weight.scala
[error]         during phase: jvm
[error]      library version: version 2.10.2-20140310-140650-2481f036a5
[error]     compiler version: version 2.10.2-20140310-140650-2481f036a5
```

**Comment by kjbrown** [comment]
Hey, 
That is a very weird error that I haven’t seen before.  Clearly the issue is that scalac can’t find a classfile that should be part of lms, but the fact that the compilation gets so far (passed OptiML/shared) before the error suggests that at least some of the lms class files do exist in the right location.  Can you check manually whether or not ForwardTransfomer.class exists? If you compile again does it always error out at the same place for the same reason?

—Kevin

**Comment by radoye** [comment]
Hey, Kevin. In the yesterday's clone the file is really missing. Looking back I did have problems with connectivity, though git should be more robust than that. 

In any case, a fresh clone of [the current head](https://github.com/stanford-ppl/hyperdsl/tree/c6c66134b5fcf1338d8fd7a83e6a504e348674e0) builds just fine. Just finished going through the [getting started walkthrough](http://stanford-ppl.github.io/Delite/optiml/getting_started.html). Ready for Delite DSL example. 

Closing, since I can't repeat the situation from yesterday. 

Thanks!
